File: backend/src/main/java/com/example/roombooking/models/UsersTable.java
package com.example.roombooking.models;

import lombok.Data;
import javax.persistence.*;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "users")
public class UsersTable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "userid")
    private int userid;

    @JsonProperty("username")
    private String username;

    @JsonProperty("email")
    private String email;

    @JsonProperty("passwordHash")
    private String passwordhash;

    @JsonProperty("roleId")
    private int roleid;

    @JsonProperty("createdAt")
    private LocalDateTime createdat;

    @JsonProperty("updatedAt")
    private LocalDateTime updatedat;

    @ManyToOne
    @JoinColumn(name = "roleid", insertable = false, updatable = false)
    private RolesTable role;
}

File: frontend/app/src/main/java/com/example/project/data/UsersTable.kt
package com.example.project.data

// UsersTable.kt
data class UsersTable(
    val userid: Int,
    val username: String,
    val email: String,
    val passwordhash: String,
    val roleid: Int,
    val createdat: String,
    val updatedat: String
)

// CreateUserRequest.kt
data class CreateUserRequest(
    val username: String,
    val email: String,
    val passwordHash: String,
    val roleId: Int
)